v3/src/components/data-display/components/legend/categorical-legend.tsx

```diff
@@ -204,16 +208,17 @@ export const CategoricalLegend = observer(
             newCatIndex = coordinatesToCatIndex(lod, numCategories, newDragPosition)
           if (newCatIndex >= 0 && newCatIndex !== dI.indexOfCategory) {
             // swap the two categories
-            duration.current = transitionDuration / 2
```

Comment:
This may seem counterintuitive, but I found removing this line made the user interaction much smoother.